src/Nethermind/Nethermind.State/OverlayWorldStateManager.cs

```diff
@@ -29,17 +29,11 @@ public class OverlayWorldStateManager(
 
     public IWorldState CreateResettableWorldState(IWorldState? forWarmup = null)
     {
-        PreBlockCaches? preBlockCaches = (forWarmup as IPreBlockCaches)?.Caches;
-        return preBlockCaches is not null
-            ? new WorldState(
-                new PreCachedTrieStore(overlayTrieStore, preBlockCaches.RlpCache),
-                _codeDb,
-                logManager,
-                preBlockCaches)
-            : new WorldState(
-                overlayTrieStore,
-                _codeDb,
-                logManager);
+        ITrieStore trieStore = overlayTrieStore;
+        if ((forWarmup as IPreBlockCaches)?.Caches is { } preBlockCaches)
+            trieStore = new PreCachedTrieStore(trieStore, preBlockCaches.RlpCache);
```

Comment:
```suggestion
        ITrieStore trieStore = (forWarmup as IPreBlockCaches)?.Caches is { } preBlockCaches ? new PreCachedTrieStore(trieStore, preBlockCaches.RlpCache) : overlayTrieStore;
```

Reply:
`trieStore` will be undefined in `new PreCachedTrieStore(trieStore, preBlockCaches.RlpCache)`